engine/servermaster/serverutil/watch_executors.go

```diff
@@ -77,6 +79,11 @@ func WatchExecutors(ctx context.Context, watcher executorWatcher, user executorI
 		case change = <-updates.C:
 		}
 
+		if change == emptyChange {
+			log.Info("receive empty executor change")
+			return cerrors.ErrExecutorWatcherClosed.GenWithStackByArgs()
+		}
```

Comment:
Can we change `case change = <-updates.C` to `case change, ok = <-updates.C`?

Reply:
addressed, this way is better.